### DISCUSSION TITLE
FUNCTION_INVOCATION_TIMEOUT on function with no change

### DISCUSSION THREAD

**Discussion by mitraraman** [vercel/community#2885] [Help] [answered]
### Page to Investigate

https://chiyoadmin.vercel.app/production/57/backlog

### Steps to Reproduce

Click 'Pull Orders'. This function:
1) does a fetch query to Shopify
2) parses the result
3) uploads into a supabase table

It is timing out without even getting to the first step.

**Comment by amyegan** [comment, marked as answer]
There may be a problem with the Shopify connection, or a delay in data processing before a result is returned. Do you see more details on function execution duration in the [Runtime Logs](https://vercel.com/docs/concepts/observability/runtime-logs)? 

If you're processing a large amount of data that takes a long time, then you might need to make use of [streaming](https://vercel.com/docs/concepts/functions/edge-functions/streaming) or work in batches. The [timeout conditions checklist](https://vercel.com/docs/concepts/functions/serverless-functions#timeout-conditions-checklist) has suggestions for a few other common causes.
